lib/manageiq/api/client/acts_as_ar_query.rb

```diff
@@ -0,0 +1,175 @@
+    class Client
+      class ActsAsArQuery
+        include Enumerable
+        attr_accessor :klass, :options
+
+        SORT_ASC = "asc".freeze
+        SORT_DESC = "desc".freeze
+
+        def initialize(klass, opts = nil)
+          @klass = klass
+          @options = opts || {}
+        end
+
+        def where(*val)
+          conditional(:where, val)
+        end
+
+        def not(*val)
+          conditional(:not, val)
+        end
+
+        def order(*val)
+          options[:order] ||= {}
+          val.flatten.each do |sval|
+            case sval
+            when String
+              sval.split(",").each do |value|
+                sort_attr, sort_order = value.split(" ")
+                options[:order][sort_attr.to_s] = sort_order || SORT_ASC
+              end
+            when Hash
+              sval.each do |sort_attr, sort_order|
+                options[:order][sort_attr.to_s] = sort_order
+              end
+            when Symbol
+              options[:order][sval.to_s] = SORT_ASC
+            end
+          end
+          self
+        end
+
+        def reorder(*val)
+          order(val)
+        end
+
+        def reverse_order
+          if options[:order].present?
+            options[:order] = options[:order].keys.each_with_object({}) { |key, hash| hash[key.to_s] = SORT_DESC }
+          end
+          self
+        end
+
+        def except(*val)
+          val.flatten.each do |key|
+            options.delete(key)
+          end
+          self
+        end
+
+        # similar to except, difference being this persists across merges.
+        def unscope(*val)
+          val.flatten.each do |key|
+            options[key] = nil
+          end
+          self
+        end
+
+        def only(*val)
+          options.slice!(val)
+          self
+        end
+
+        def offset(val)
+          assign_arg :offset, val
+        end
+
+        def limit(val)
+          assign_arg :limit, val
+        end
+
+        def select(*args)
+          append_hash_arg :select, *args
+        end
+
+        def to_a
+          @results ||= klass.ar_search(options)
+        end
+
+        def all
+          self
+        end
+
+        def count(*_args)
+          to_a.size
+        end
+
+        def first
+          positional(:first)
+        end
+
+        def second
+          positional(:second)
+        end
+
+        def third
+          positional(:third)
+        end
+
+        def fourth
+          positional(:fourth)
+        end
+
+        def fifth
+          positional(:fifth)
+        end
+
+        def last
+          positional(:last)
+        end
+
+        def ids
+          # Change to using .collect(&:id) once PR# 23 is merged
+          to_a.collect { |resource| resource.data["id"] }
+        end
+
+        delegate :size, :take, :each, :empty?, :presence, :to => :to_a
+
+        private
+
+        def positional(position)
+          defined?(@results) ? @results.send(position) : to_a.send(position)
```

Comment:
nitpick (you can ignore)

since `to_a` already handles the cached example, can you just send all these to `to_a`?

e.g.:

```ruby
def positional(position)
  to_a.send(position)
end
```

and if that is the case, just delegate all of these